smslibrary/src/main/java/com/eis/communication/MessageHandler.java

```diff
@@ -4,23 +4,26 @@
  * Class used to switch from actual communication data to Message and back
  * Should use a <a href='https://en.wikipedia.org/wiki/Singleton_pattern'>Singleton pattern</a>
  *
+ * @param <D> The data type used by the channel
+ * @param <P> The peer type used by the channel
  * @param <M> The type of message to parse and output
  */
-public interface MessageHandler<M extends Message> {
+public interface MessageHandler<D,P,M extends Message> {
 
     /**
-     * Interprets a string arrived via the communication channel and parses it to a message
+     * Interprets data arrived via the communication channel and parses it to a message
      *
-     * @param data data from the communication channel
-     * @return the message if the string has been parsed correctly, null otherwise
+     * @param peerData from the communication channel
+     * @param data from the communication channel
+     * @return the message if the data has been parsed correctly, null otherwise
      */
-    M parseMessage(String data, String peerData);
+    M parseMessage(P peerData, D data);
 
     /**
-     * Translates a message into a string that can be sent via communication channel
+     * Translates a message into data that can be sent via communication channel
      *
      * @param message message to be translated
```

Comment:
I think it's better don't repeat the same word _message_

Reply:
You're correct, I'll fix this